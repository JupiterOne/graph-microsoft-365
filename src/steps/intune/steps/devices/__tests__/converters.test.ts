// import { RelationshipDirection } from '@jupiterone/integration-sdk-core';

import {
  createManagedDeviceEntity,
  createUserDeviceDirectRelationship,
  createUserDeviceMappedRelationship,
} from '../converters';
import exampleManagedDevice from './fixtures/exampleManagedDevice';
import exampleManagedDeviceEntity from './fixtures/exampleManagedDeviceEntity';

describe('createManagedDeviceEntity', () => {
  test('properties transferred', () => {
    expect(createManagedDeviceEntity(exampleManagedDevice)).toMatchSnapshot(
      'createManagedDeviceEntity',
    );
  });
});

describe('createUserDeviceMappedRelationship', () => {
  test('returns undefined without userId or email', () => {
    expect(
      createUserDeviceMappedRelationship(exampleManagedDeviceEntity),
    ).toBeUndefined();
  });
  test('properties transferred with userId', () => {
    expect(
      createUserDeviceMappedRelationship(exampleManagedDeviceEntity, 'userId'),
    ).toMatchSnapshot('createUserDeviceMappedRelationship');
  });
  test('properties transferred with email', () => {
    expect(
      createUserDeviceMappedRelationship(
        exampleManagedDeviceEntity,
        undefined,
        'email',
      ),
    ).toMatchSnapshot('createUserDeviceMappedRelationship');
  });
  test('properties transferred with both userId and email', () => {
    expect(
      createUserDeviceMappedRelationship(
        exampleManagedDeviceEntity,
        'userId',
        'email',
      ),
    ).toMatchSnapshot('createUserDeviceMappedRelationship');
  });
});

describe('createUserDeviceDirectRelationship', () => {
  test('properties transferred', () => {
    expect(
      createUserDeviceDirectRelationship(exampleManagedDeviceEntity, {
        _class: 'USER',
        _key: '1234',
        _type: 'microsoft_365_user',
      }),
    ).toMatchSnapshot('createUserDeviceDirectRelationship');
  });
});
