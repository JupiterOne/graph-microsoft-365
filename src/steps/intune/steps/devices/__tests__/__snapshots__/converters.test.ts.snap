// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createManagedDeviceEntity properties transferred: createManagedDeviceEntity 1`] = `
Object {
  "BYOD": true,
  "_class": Array [
    "Device",
  ],
  "_key": "11111111-1111-1111-1111-111111111111",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "activationLockBypassCode": null,
        "androidSecurityPatchLevel": "",
        "azureADDeviceId": "11111111-1111-1111-1111-111111111111",
        "azureADRegistered": null,
        "complianceGracePeriodExpirationDateTime": "2021-02-17T19:13:41Z",
        "complianceState": "noncompliant",
        "configurationManagerClientEnabledFeatures": null,
        "deviceActionResults": Array [],
        "deviceCategoryDisplayName": "",
        "deviceEnrollmentType": "deviceEnrollmentManager",
        "deviceHealthAttestationState": null,
        "deviceName": "Dude’s iPhone",
        "deviceRegistrationState": "registered",
        "easActivated": false,
        "easActivationDateTime": "0001-01-01T00:00:00Z",
        "easDeviceId": "ASDFASDFASDFASDFASDFASDF",
        "emailAddress": "dude@email.com",
        "enrolledDateTime": "2021-02-17T16:24:45Z",
        "exchangeAccessState": "none",
        "exchangeAccessStateReason": "none",
        "exchangeLastSuccessfulSyncDateTime": "0001-01-01T00:00:00Z",
        "freeStorageSpaceInBytes": 15008268288,
        "id": "11111111-1111-1111-1111-111111111111",
        "imei": "1234567890",
        "isEncrypted": true,
        "isSupervised": false,
        "jailBroken": "False",
        "lastSyncDateTime": "2021-02-23T08:25:25Z",
        "managedDeviceName": "dudes_IPhone_2/17/2021_4:24 PM",
        "managedDeviceOwnerType": "personal",
        "managementAgent": "mdm",
        "manufacturer": "Apple",
        "meid": "35324910401736",
        "model": "iPhone 11 Pro",
        "operatingSystem": "iOS",
        "osVersion": "14.4",
        "partnerReportedThreatState": "unknown",
        "phoneNumber": "******5480",
        "remoteAssistanceSessionErrorDetails": null,
        "remoteAssistanceSessionUrl": null,
        "serialNumber": "1234567890",
        "subscriberCarrier": "Sprint",
        "totalStorageSpaceInBytes": 68719476736,
        "userDisplayName": "",
        "userId": "userId",
        "userPrincipalName": "Dude",
        "wiFiMacAddress": "84ad8db5b351",
      },
    },
  ],
  "_type": "intune_managed_device",
  "assetTag": "11111111-1111-1111-1111-111111111111",
  "category": "endpoint",
  "complianceState": "noncompliant",
  "createdOn": undefined,
  "deviceName": "Dude’s iPhone",
  "displayName": "Dude’s iPhone",
  "easDeviceId": false,
  "encrypted": true,
  "hardwareModel": "iPhone 11 Pro",
  "hardwareSerial": "1234567890",
  "hardwareVendor": "Apple",
  "hardwareVersion": "iPhone 11 Pro",
  "iccid": undefined,
  "id": "11111111-1111-1111-1111-111111111111",
  "imei": "1234567890",
  "jailBroken": false,
  "make": "Apple",
  "managed": undefined,
  "managementAgent": "mdm",
  "meid": "35324910401736",
  "model": "iPhone 11 Pro",
  "name": "Dude’s iPhone",
  "osDetails": "iOS version 14.4",
  "osName": "iOS",
  "osVersion": "14.4",
  "ownerType": "personal",
  "phoneNumber": "******5480",
  "physical": true,
  "registrationState": "registered",
  "serial": "1234567890",
  "serialNumber": "1234567890",
  "state": undefined,
  "supervised": false,
  "udid": undefined,
  "userDisplayName": "",
  "userEmails": Array [
    "dude@email.com",
  ],
  "userId": "userId",
  "username": "Dude",
  "version": "iPhone 11 Pro",
  "wifiMacAddress": "84ad8db5b351",
}
`;

exports[`createUserDeviceDirectRelationship properties transferred: createUserDeviceDirectRelationship 1`] = `
Object {
  "_class": "HAS",
  "_fromEntityKey": "1234",
  "_key": "1234|has|11111111-1111-1111-1111-111111111111",
  "_toEntityKey": "11111111-1111-1111-1111-111111111111",
  "_type": "microsoft_365_user_has_intune_managed_device",
  "displayName": "HAS",
}
`;

exports[`createUserDeviceMappedRelationship properties transferred with both userId and email: createUserDeviceMappedRelationship 1`] = `
Object {
  "_class": "HAS",
  "_key": "11111111-1111-1111-1111-111111111111|has|userId",
  "_mapping": Object {
    "relationshipDirection": "REVERSE",
    "sourceEntityKey": "11111111-1111-1111-1111-111111111111",
    "targetEntity": Object {
      "_class": "User",
      "_key": "userId",
      "_type": "microsoft_365_user",
      "email": "email",
      "id": "userId",
    },
    "targetFilterKeys": Array [
      Array [
        "_class",
        "id",
      ],
      Array [
        "_class",
        "email",
      ],
    ],
  },
  "_type": "microsoft_365_user_has_intune_managed_device",
  "displayName": "HAS",
}
`;

exports[`createUserDeviceMappedRelationship properties transferred with email: createUserDeviceMappedRelationship 1`] = `
Object {
  "_class": "HAS",
  "_key": "11111111-1111-1111-1111-111111111111|has|REVERSE:_class=User:id=undefined:_class=User:email=email",
  "_mapping": Object {
    "relationshipDirection": "REVERSE",
    "sourceEntityKey": "11111111-1111-1111-1111-111111111111",
    "targetEntity": Object {
      "_class": "User",
      "_key": undefined,
      "_type": "microsoft_365_user",
      "email": "email",
      "id": undefined,
    },
    "targetFilterKeys": Array [
      Array [
        "_class",
        "id",
      ],
      Array [
        "_class",
        "email",
      ],
    ],
  },
  "_type": "microsoft_365_user_has_intune_managed_device",
  "displayName": "HAS",
}
`;

exports[`createUserDeviceMappedRelationship properties transferred with userId: createUserDeviceMappedRelationship 1`] = `
Object {
  "_class": "HAS",
  "_key": "11111111-1111-1111-1111-111111111111|has|userId",
  "_mapping": Object {
    "relationshipDirection": "REVERSE",
    "sourceEntityKey": "11111111-1111-1111-1111-111111111111",
    "targetEntity": Object {
      "_class": "User",
      "_key": "userId",
      "_type": "microsoft_365_user",
      "email": undefined,
      "id": "userId",
    },
    "targetFilterKeys": Array [
      Array [
        "_class",
        "id",
      ],
      Array [
        "_class",
        "email",
      ],
    ],
  },
  "_type": "microsoft_365_user_has_intune_managed_device",
  "displayName": "HAS",
}
`;
